pkg/gencred/cmd/main.go

```diff
@@ -0,0 +1,210 @@
+		printErrAndExit(fmt.Sprintf("-o, --output already exists and is a directory: %v.", output), 1)
+	}
+
+	if serviceaccount && certificate {
+		printErrAndExit("-c, --certificate and -s, --serviceaccount are mutually exclusive options.", 1)
+	}
+}
+
+// printErr prints error to stderr.
+func printErr(msg string) {
+	_, _ = fmt.Fprintln(os.Stderr, msg)
+}
+
+// printErrAndExit prints error to stderr and exits with te provided code.
+func printErrAndExit(msg string, code int) {
+	printErr(msg)
+	os.Exit(code)
+}
+
+// dirExists checks if a path exists and is a directory.
+func dirExists(path string) bool {
+	info, err := os.Stat(path)
+	if os.IsNotExist(err) {
+		return false
+	}
+	return info.IsDir()
+}
+
+// fileExists checks if a path exists and is a file.
+func fileExists(path string) bool {
+	info, err := os.Stat(path)
+	if os.IsNotExist(err) {
+		return false
+	}
+	return !info.IsDir()
+}
+
+// mergeConfigs merges an existing kubeconfig file with a new entry with precedence given to the existing config.
+func mergeConfigs(kubeconfig *[]byte) {
+	tmpFile, err := ioutil.TempFile(os.TempDir(), "")
+	if err != nil {
+		printErrAndExit(err.Error(), 1)
+	}
+	defer os.Remove(tmpFile.Name())
+
+	err = ioutil.WriteFile(tmpFile.Name(), *kubeconfig, 0644)
+	if err != nil {
+		printErrAndExit(err.Error(), 1)
+	}
+
+	loadingRules := clientcmd.ClientConfigLoadingRules{
+		Precedence: []string{tmpFile.Name(), output},
+	}
+
+	mergedConfig, err := loadingRules.Load()
+	if err != nil {
+		printErrAndExit(err.Error(), 1)
+	}
+
+	json, err := runtime.Encode(latest.Codec, mergedConfig)
+	if err != nil {
+		printErrAndExit(err.Error(), 1)
+	}
+
+	*kubeconfig, err = yaml.JSONToYAML(json)
+	if err != nil {
+		printErrAndExit(err.Error(), 1)
+	}
+}
+
+// writeConfig writes a kubeconfig file to an output file.
+func writeConfig(clientset kubernetes.Interface) {
+	// kubeconfig is a kubernetes config.
+	var kubeconfig []byte
+
+	dir, file := filepath.Split(output)
+
+	err := os.MkdirAll(dir, os.ModePerm)
+	if err != nil {
+		printErrAndExit(fmt.Sprintf("unable to create output directory %v: %v.", dir, err), 1)
+	}
+
+	if certificate {
+		if kubeconfig, err = gencred.CreateKubeConfigWithCertificateCredentials(clientset, name); err != nil {
+			printErrAndExit(fmt.Sprintf("unable to create kubeconfig file with cert and key for %v: %v.", name, err), 1)
+		}
+	} else {
+		// Service account credentials are the default if unspecified.
+		if kubeconfig, err = gencred.CreateKubeConfigWithServiceAccountCredentials(clientset, name); err != nil {
+			printErrAndExit(fmt.Sprintf("unable to create kubeconfig file with service account for %v: %v.", name, err), 1)
+		}
+	}
+
+	if !overwrite && fileExists(output) {
+		mergeConfigs(&kubeconfig)
+	}
+
+	err = ioutil.WriteFile(output, kubeconfig, 0644)
+	if err != nil {
+		printErrAndExit(fmt.Sprintf("unable to write to file %v: %v.", file, err), 1)
+	}
+
+}
+
+// init entry point.
+func init() {
+
+	// --context
```

Comment:
Can you move combine this with the next line, and get rid of all the extra line breaks?